### ISSUE TITLE
Feature: Pass event data to Apple script

### ISSUE THREAD

**Issue by luisherranz** [leits/MeetingBar#373]
**Is your feature request related to a problem? Please describe.**

I'd like to use an Apple script to do different things depending on the event that is starting. For example, I'd like to turn on the Do Not Disturb mode when the event name is "Focus".

**Describe the solution you'd like**

It'd be great if the Apple script could access the data of the current event using a variable injected by MeetingBar.

```applescript
if (summary of MeetingBarEvent = "Focus") then
  set _start to start date of MeetingBarEvent
  set _end to end date of MeetingBarEvent
  set _duration to -(_start - _end) / 60

  display dialog "The name of the event is: " & summary of MeetingBarEvent
  display dialog "Duration in minutes is: " & _duration
end if
```

**Describe alternatives you've considered**

It's possible to get this information from the Calendar app, but it needs to be opened to get the information, which is very inconvenient if you don't use it at all.

Also, it adds unnecessary complexity. Most of the time you want your script to act based on the current event, not on other events.

```applescript
set _now to current date

tell application "Calendar"
	tell calendar "YOUR CALENDAR NAME"
		set _list to (every event whose (start date < _now) and (end date > _now))
		
		repeat with _event in _list
			if (summary of _event as string = "Focus") then
				set _start to start date of _event
				set _end to end date of _event
				set _duration to -(_start - _end) / 60

				display dialog "The name of the event is: " & summary of _event
				display dialog "Duration in minutes is: " & _duration
			end if
		end repeat
	end tell
end tell
```

**Comment by leits** [comment]
Hi @luisherranz,
Thanks to @jgoldhammer and his changes in #166 from version 4.0 you can!
Here is an example of how to use it:
```
# the method to be called with the following parameters for the next meeting.
#
# 1. parameter - eventId (string) - unique identifier from apples eventkit implementation
# 2. parameter - title (string) - the title of the event (event title can be null, although it makes no sense!)
# 3. parameter - allday (bool) - true for allday events, false for non allday events
# 4. parameter - startDate (date) - needs casting in apple script to output (e.g. startDate as text)
# 5 .parameter - endDate (date) - needs casting in apple script to output (e.g. startDate as text)
# 6. parameter - eventLocation (string) - if no location is set, the value will be "EMPTY"
# 7. parameter - repeatingEvent (bool) - true if it is part of an repeating event, false for single event
# 8. parameter - attendeeCount (int32) - number of attendees- 0 for events without attendees
# 9. parameter - meetingUrl (string) - the url to the meeting found in notes, url or location - only one meeting url is supported - if no meeting url is set, the value will be "EMPTY"
# 10. parameter - meetingService (string), e.g MS Teams or Zoom- if no meeting service is found, the meeting service value is "EMPTY"
# 11. parameter - meetingNotes (string)- the complete notes of a meeting -  if no notes are set, value "EMPTY" will be used

on meetingStart(eventId, title, allday, startDate, endDate, eventLocation, repeatingEvent, attendeeCount, meetingUrl, meetingService, meetingNotes)
   tell application "Finder"
       activate
       display dialog title buttons {"OK"} default button "OK"
   end tell
end meetingStart
```


cc: @sfdye, @SikhNerd, @hunleyd, @mattmink, @slawekp

**Comment by SikhNerd** [comment]
This looks great! @leits  I'm assuming you mean version 5.0 (yet to be released?)

**Comment by hunleyd** [comment]
> This looks great! @leits I'm assuming you mean version 5.0 (yet to be released?)

the [commit](https://github.com/leits/MeetingBar/commit/819c4afcd10246ab511dc3fb80073a226a03cd88) landed in 4.2.1 @SikhNerd 
i'm also please to report it appears to be working here
